lib/grape/endpoint.rb

```diff
@@ -232,6 +232,7 @@ def run(env)
 
       route_setting(:saved_validations).each do |validator|
         begin
+          validator.context = Hashie::Mash.new(request: request, cookies: cookies)
```

Comment:
Hmmm... The request will be GCed with validator object I think. We can make context nil after validation but it's a bit over engineering the thing.
Do you see here any use case/example that could lead to memory leak/stack overflow?